### ISSUE TITLE
PaddingBottom on Camera fitBounds does not work anymore

### ISSUE THREAD

**Issue by fredbralvalex** [react-native-mapbox-gl/maps#841]
**Describe the bug**
Using this.camera.fitBounds([lng, lat], [lng, lat], [0, 0, bottom, 0], 1000), the bottom padding attribute work as Vertical padding ignoring the 0 for top padding and using the same value as bottom.

**To Reproduce**
Calling this.camera.fitBounds([lng, lat], [lng, lat], [top, right, bottom, left], 1000) setting the bottom padding by half screen height and 0 (or any other value) for top, right, left.

Example:

```js

	selectRoute = () => {
		...
		this.fitBounds( [highLong, highLat], [lowerLong, lowerLat], [0, 0, height*0.5, 0]);
	}

	fitBounds(northEast: SingleCoordinate, southWest: SingleCoordinate, padding: Array<number> = [50,50,50,50]) {
		this.camera.fitBounds(northEast, southWest, padding, 500);
	}

	render() {
		return (
			<MapboxGL.Camera
				ref={camera => this.camera = camera}
				followUserLocation={false}
				followUserMode={'normal'}
			/>
		)
	}
	
```

**Expected behavior**
The center of northEastCoordinates and southWestCoordinates used must be at the center of the upper half of the screen (padding half height). However it is centered in the middle of screen.

**Versions (please complete the following information):**

Platform: Android, iOS 13.3.1
Device: Iphone XR
react-native-mapbox-gl Version 8.0.0-rc1 (8.0.0)*
React Native Version 0.61.5
Mapbox Version 1.0.0-beta10

* It works fine  with react-native-mapbox-gl Version 7.2.0

**Comment by mfazekas** [comment]
The upstream issue causing it: https://github.com/mapbox/mapbox-gl-native-ios/issues/198

**Comment by baijii** [comment]
@mfazekas Are you sure, this is causing the problem? The top padding is ignored in iOS as well as on Android devices.

**Comment by mfazekas** [comment]
@baijii pls check this comment: https://github.com/mapbox/mapbox-gl-native-ios/issues/198#issuecomment-620387699

I think top padding is not ignored, it's just looks it considered as a minimum padding and it's always made symmetrical. 

And sure this issue is likely coming from the C++ layer which is common in ios and android.

**Comment by baijii** [comment]
@mfazekas My bad, thanks for explaining

**Comment by mfazekas** [comment]
Example to reproduce the issue:

```es6
import React from 'react';
import {SafeAreaView, Button} from 'react-native';
import MapboxGL, {
  MapView,
  ShapeSource,
  LineLayer,
  Camera,
} from '@react-native-mapbox-gl/maps';

import accessToken from './accesstoken.json';

MapboxGL.setAccessToken(accessToken);

const aLine = {
  type: 'LineString',
  coordinates: [[-74.00597, 40.71427], [-74.00697, 40.71527]],
};

class BugReportExample extends React.Component {
  render() {
    return (
      <SafeAreaView style={{flex: 1, flexDirection: 'column'}}>
        <Button
          title="fitBounds"
          onPress={() => {
            this.camera.fitBounds(
              [-74.00597, 40.71427],
              [-74.00697, 40.71527],
              [0, 0, 300, 0],
            );
          }}
        />
        <MapView style={{flex: 1}}>
          <Camera
            centerCoordinate={[-74.00597, 40.71427]}
            zoomLevel={14}
            ref={ref => {
              this.camera = ref;
            }}
          />
          <ShapeSource id="idStreetLayer" shape={aLine}>
            <LineLayer id="idStreetLayer" />
          </ShapeSource>
        </MapView>
      </SafeAreaView>
    );
  }
}

export default BugReportExample;

```

**Comment by mfazekas** [comment]
On ios as a workaround, add:

```ruby
ENV['REACT_NATIVE_MAPBOX_MAPBOX_IOS_VERSION'] = '~> 5.6.0'
```

example `Podfile`:


```ruby
platform :ios, '9.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

ENV['REACT_NATIVE_MAPBOX_MAPBOX_IOS_VERSION'] = '~> 5.6.0'
...
```

**Comment by mfazekas** [comment]
Also submitted fix to https://github.com/mapbox/mapbox-gl-native-ios/pull/323

**Comment by sunny-lirr** [comment]
I installed 8.1.0rc2, used the suggested fix on iOS, and tested the example code above. On both platforms, the fit bounds button worked on the first press.

I then pressed the button again without moving the map. On iOS, nothing happened, which seems like the intended behavior. On Android, the map moved up a bit, then an error appeared:

![image](https://user-images.githubusercontent.com/53345676/86848989-007f2900-c07d-11ea-9c1a-78bcaf3bd706.png)

**Comment by cglacet** [comment]
I experience the same bug as @sunny-lirr (only on Android too). I'm not using fitBounds, but I'm updating `bounds` state that I use as the map's camera `bounds` props. I also use a constant padding. 


On high zooms, for example with the following bounds: 

```js 
const bounds = {
    sw: [2.310149073600769, 48.85784],
    ne: [2.335765, 48.869005749964536],
    paddingLeft: 10,
    paddingRight: 10,
    paddingTop: 100,
    paddingBottom: 100,
};
```

On Android, passing this props to the camera give the above mentioned crash: 

```js
<Mapbox.Camera
    animationMode={'flyTo'}
    animationDuration={this.state.animationDuration}
    bounds={bounds}
/>
```

On my side the bug only happens on high zoom values but I can still zoom up to the requested level manually (fitting the input bounds manually). I guess that's the same bug appearing when calling the `fitBounds` method.

**Comment by mfazekas** [comment]
@cglacet this should be addressed by #973, can you please verify?!

https://github.com/react-native-mapbox-gl/maps/issues/956#issuecomment-665886930

**Comment by cglacet** [comment]
I can, do I need to compile anything once I've updated the source? Or should I just re-build my react-native application?

**Comment by mfazekas** [comment]
@cglacet it's a native part, so you'll need to invoke the gradle stuff (via gralew or `android studio` or via `react-native run-android`)

Thanks for testing.

**Comment by luskin** [comment]
Does anybody have a solution for this? It's very weird for us. Padding up to 40 works fine, but once we get to 40 we receive the error:

```
Mapbox error [event]:General [code]:-1 [message]:Unable to calculate appropriate zoom level for bounds. Vertical or horizontal padding is greater than map's height or width. {"filePath": "virtual bool mbgl::MGLCoreLoggingObserver::onRecord(mbgl::EventSeverity, mbgl::Event, int64_t, const std::string &)", "level": "error", "line": 30, "message": "[event]:General [code]:-1 [message]:Unable to calculate appropriate zoom level for bounds. Vertical or horizontal padding is greater than map's height or width."}
```

Our implementation for this scenario is quite simple:

```typescript
     <MapboxGL.MapView ref={mapViewRef} style={styles.map} styleURL={styleUrl}>
        <MapboxGL.Camera
          ref={mapCameraRef}
          defaultSettings={{
            bounds: bounds
              ? {
                  ne: bounds?.ne ?? 0,
                  sw: bounds?.sw ?? 0,
                  paddingLeft: 40,
                  paddingRight: 40,
                  paddingTop: 40,
                  paddingBottom: 40,
                }
              : undefined,
          }}
        />
        {children}
      </MapboxGL.MapView>
```

**Comment by ferdicus** [comment]
Addressed in #1057

**Comment by yash-vardhan-tyagi** [comment]
> Can be used like this also, as soon as the screen renders the effects will be visible:
> 
> ```js
>const cameraRef : any = useRef(null);
>
>  useEffect(() => {
>    if (originCoords.length && destination) {
>      // Check if cameraRef is initialized
>      if (cameraRef.current) {
>        // Access the camera instance through cameraRef.current
>        cameraRef.current.fitBounds(
>          [originLng, originLat],
>          [destinationLng, destinationLat],
>          [60, 120, 60, 120],3000
>        );
>      }
>    }
>  }, [originCoords, destination, originLng, originLat, destinationLng, destinationLat]); import React from 'react';
>
> <MapLibreGL.Camera
>              ref={cameraRef}
>              centerCoordinate={region}
>              pitch={50}
>              animationMode='flyTo'
>              animationDuration={6000} 
>              />              
>
>
> ```
